test/handlers/delete-small-catch-handler.unit.test.js

```diff
@@ -0,0 +1,172 @@
+const mockGetByIdSmallCatch = jest.fn()
+const mockDoMap = jest.fn()
+const mockDeleteById = jest.fn()
+const mockGetByIdSubmission = jest.fn()
+const mockGetFromLink = jest.fn()
+const mockIsAllowedParam = jest.fn()
+const mockTestLocked = jest.fn()
+const mockTextFromNum = jest.fn()
+
+const DeleteSmallCatchHandler = require('../../src/handlers/delete-small-catch')
+const { getMockH } = require('../test-utils/server-test-utils')
+
+jest.mock('../../src/api/small-catches', () => {
+  return jest.fn().mockImplementation(() => ({
+    getById: mockGetByIdSmallCatch,
+    doMap: mockDoMap,
+    deleteById: mockDeleteById
+  }))
+})
+
+jest.mock('../../src/api/submissions', () => {
+  return jest.fn().mockImplementation(() => ({
+    getById: mockGetByIdSubmission
+  }))
+})
+
+jest.mock('../../src/api/activities', () => {
+  return jest.fn().mockImplementation(() => ({
+    getFromLink: mockGetFromLink
+  }))
+})
+
+jest.mock('../../src/handlers/common', () => ({
+  testLocked: mockTestLocked,
+  isAllowedParam: mockIsAllowedParam,
+  monthHelper: {
+    find: {
+      textFromNum: mockTextFromNum
+    }
+  }
+}))
+
+describe('delete-small-catch-handler.unit', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const getMockRequest = (overrides) => ({
+    params: { id: '123' },
+    path: '/delete-small-catch',
+    cache: jest.fn(() => ({
+      get: jest.fn().mockResolvedValue({
+        submissionId: 'sub1',
+        licenceNumber: 'LN123',
+        postcode: 'AB12CD',
+        year: '2025'
+      }),
+      set: jest.fn().mockResolvedValue()
+    })),
+    ...overrides
+  })
+
+  describe('doGet', () => {
+    it('should throw ResponseError if id param is not allowed', async () => {
+      const request = getMockRequest()
+      const h = getMockH()
+      mockIsAllowedParam.mockReturnValue(false)
```

Comment:
Same thing as the other PR (since these are so similar) – would prefer to mock values only once unless we have a reason not to.